common/auth.go

```diff
@@ -15,6 +15,41 @@
 
 package common
 
-type AuthBackend interface {
-	
-}
\ No newline at end of file
+type Role interface {
+	Name() string
+}
+
+type role struct {
+	name string
+}
+
+// Represents the type of credential (e.g., certificate,
+// username-password, etc.
+type CredentialType string
+
+const (
+	CredentialUsernamePassword = "userPass"
+	CredentialNone             = "none"
+)
+
+// Container for various credentials. Currently containing Username/Password
+// but keys, certificates, etc. can be used in the future.
+type Credential struct {
+	Type     CredentialType
+	Username string
+	Password string
+}
+
+// MakeCredentialFromCliArgs takes all possible CLI
```

Comment:
I think there is this idea that providing passwords as command line parameters is bad form, since everyone on the system can see your password in clear text when they do "ps auxw". A preferred way - I believe - is to specify passwords as an environment variable to the process. You have to be root to examine a running process' environment, so that's a bit more secure than putting it on the command line.

Maybe we could add looking for a specific environment variable, if the password cannot be found on the command line?